packages/iris-grid/src/sidebar/TableCsvExporter.tsx

```diff
@@ -184,12 +201,27 @@ class TableCsvExporter extends Component<
     const snapshotRanges = this.getSnapshotRanges();
     if (this.validateOptionInput()) {
       onDownloadStart();
-      model.export().then(frozenTable => {
-        const tableSubscription = frozenTable.setViewport(0, 0);
-        tableSubscription.getViewportData().then(() => {
-          onDownload(fileName, frozenTable, tableSubscription, snapshotRanges);
+      model
+        .export()
+        .then(frozenTable => {
+          const tableSubscription = frozenTable.setViewport(0, 0);
+          tableSubscription
+            .getViewportData()
+            .then(() => {
+              onDownload(
+                fileName,
+                frozenTable,
+                tableSubscription,
+                snapshotRanges
+              );
+            })
+            .catch(error => {
+              this.setDownloadPromiseError(error);
+            });
+        })
+        .catch(error => {
+          this.setDownloadPromiseError(error);
```

Comment:
This could probably be rewritten as an async/await function, and it'd be a bit cleaner, you'd only need one catch statement I think then.

```
try {
   const frozenTable = await model.export();
   const tableSubscription = frozenTable.setViewport(0, 0);
   const viewportData = await tableSubscription.getViewportData();
   onDownload(...)
} catch (e) {
   ...
}
```